src/featurelib.cpp

```diff
@@ -1,4 +1,49 @@
-#include "featurelib.h"
+#include "FeatureLib.h"
 
-featurelib::featurelib() {
+FeatureLib::FeatureLib(const std::vector<std::uint8_t> &inputVector, int Width, int Height) {
+  Image = VectorToMat(inputVector, Height, Width);
+  cv::cvtColor( Image, Image, cv::COLOR_BGR2GRAY );
 }
+
+cv::Mat FeatureLib::VectorToMat(const std::vector<std::uint8_t> &inputVector, int Height, int Width) {
+  return cv::Mat(Height, Width, CV_8UC1, (std::uint8_t *) inputVector.data());
+}
+
+std::vector<uint8_t> FeatureLib::MatToVector(const cv::Mat &mat) {
+  std::vector<uint8_t> result;
+  if (mat.isContinuous()) {
+    result.assign((uint8_t *) mat.datastart, (uint8_t *) mat.dataend);
+  } else {
+    for (int i = 0; i < mat.rows; ++i) {
+      result.insert(result.end(), (uint8_t *) mat.ptr<uchar>(i), (uint8_t *) mat.ptr<uchar>(i) + mat.cols);
+    }
+  }
+  return result;
+}
+
+std::vector<libstructs::KeyPoint> FeatureLib::ConvertToStructure(const std::vector<cv::KeyPoint> &CvKeyPointsVec) {
+  std::vector<libstructs::KeyPoint> output;
```

Comment:
I think, it would be more efficient to call [`vector.reserve()`](https://en.cppreference.com/w/cpp/container/vector/reserve) before doing the push backs as you know the size.

Reply:
done
but don't know how to get number of structure fields by method, or something like that

Reply:
Sorry, I don't understand the question